pubnub-kotlin/pubnub-kotlin-core-api/src/commonMain/kotlin/com/pubnub/api/models/consumer/objects/membership/MembershipInclude.kt

```diff
@@ -0,0 +1,99 @@
+package com.pubnub.api.models.consumer.objects.membership
+
+import com.pubnub.api.models.consumer.objects.member.Include
+import kotlin.jvm.JvmStatic
+
+class MembershipInclude(
+    override val includeCustom: Boolean = false,
+    override val includeStatus: Boolean = false,
+    override val includeType: Boolean = false,
+    override val includeTotalCount: Boolean = false,
+    val includeChannel: Boolean = false,
+    val includeChannelCustom: Boolean = false,
+    val includeChannelType: Boolean = false,
+    val includeChannelStatus: Boolean = false,
+) : Include() {
+    companion object {
+        @JvmStatic
+        fun includeCustom(includeCustom: Boolean): Builder {
+            return Builder().includeCustom(includeCustom)
+        }
+
+        @JvmStatic
+        fun includeStatus(includeStatus: Boolean): Builder {
+            return Builder().includeStatus(includeStatus)
+        }
+
+        @JvmStatic
+        fun includeType(includeType: Boolean): Builder {
+            return Builder().includeType(includeType)
+        }
+
+        @JvmStatic
+        fun includeTotalCount(includeTotalCount: Boolean): Builder {
+            return Builder().includeTotalCount(includeTotalCount)
+        }
+
+        @JvmStatic
+        fun includeChannel(includeChannel: Boolean): Builder {
+            return Builder().includeChannel(includeChannel)
+        }
+
+        @JvmStatic
+        fun includeChannelCustom(includeChannelCustom: Boolean): Builder {
+            return Builder().includeChannelCustom(includeChannelCustom)
+        }
+
+        @JvmStatic
+        fun includeChannelType(includeChannelType: Boolean): Builder {
+            return Builder().includeChannelType(includeChannelType)
+        }
+
+        @JvmStatic
+        fun includeChannelStatus(includeChannelStatus: Boolean): Builder {
+            return Builder().includeChannelStatus(includeChannelStatus)
+        }
+    }
```

Comment:
not sure if this is needed, I think `fun builder(): Builder` is enough for an entry point

moreover, I don't think Kotlin users need the builder at all,
maybe experiment with `@kotlin.Deprecated(message = "JUST FOR JAVA", level = DeprecationLevel.HIDDEN)` to hide it from Kotlin but allow Java usage and see if it works?

otherwise we could have two different class hierarchies in the Kotlin and Java modules (in `com.pubnub.java.*` package)

Reply:
another thing you could try:

add a no-argument secondary constructor to `MembershipInclude` with DeprecationLevel.HIDDEN
move the methods like `fun includeCustom` onto `MembershipInclude` with DeprecationLevel.HIDDEN
remove the `Builder` class and `build()` method entirely

now from Java it should be possible to do `new MembershipInclude().includeCustom(...)` (I think?) 

the upside is that you won't have to repeat all the properties in Builder again and their default values

Reply:
Ok, as discussed.